Sources/MongoSwift/SDAM.swift

```diff
@@ -1,14 +1,15 @@
 import Foundation
 import mongoc
 
-/// A struct representing a server connection, consisting of a host and port.
-public struct ConnectionId: Equatable {
-    /// A string representing the host for this connection.
+/// A struct representing a server address, consisting of a host and port.
+public struct Address: Equatable {
```

Comment:
nit: let's either call it `ServerAddress` or correct the comment to explain this is a generic address. Also, now that we have NIO as a dependency, is there a good reason we aren't just using their [SocketAddress](https://github.com/apple/swift-nio/blob/master/Sources/NIO/SocketAddresses.swift)? It seems to serve a similar purpose.

Reply:
done. I added a `try!` when converting from a `mongoc_host_list_t` to a `SocketAddress`, since I figured libmongoc would only contain well-formatted addresses in its server descriptions. Is that a safe assumption to make?

Reply:
`SocketAddress` only works on resolved IP addresses, so things like "localhost" or a domain name won't be able to be stored in it, which I think doesn't work for us. As such, I've reverted the commits that enabled its usage in the driver.

Reply:
I've updated the comment to say a "network address" instead of "server address"